### PR TITLE
PoC: updates indexed from state [KVL-916]

### PR DESCRIPTION
**Pull request by miklos-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/reindexfromstate/StateToUpdateMapping.scala [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        contract,
+        contractInstance,
+        submissionSeed,
+      ),
+      transactionId = TransactionId.assertFromString(
+        UUID.randomUUID().toString
+      ), // TODO: Can be determined from state?
+      recordTime = recordTime(),
+      divulgedContracts = List.empty, // No divulged contracts at creation.
+      blindingInfo = Some(blindingInfoFromContractState(contract, contractId)),
```
Comment (meiersi-da) [inline ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/reindexfromstate/StateToUpdateMapping.scala:155]: @miklos-da : I believe you can save code by setting this to `None`, as the indexer will then compute it by itself from the `CommittedTransaction`

**Review by meiersi-da — COMMENTED** [review]
Thank you @miklos-da for this very valuable PoC. It provides clarity for all kinds of data except contract instances. For these, I'd suggest to see whether the simplification outlined in the comments below work. 

Once we have clarity for contract instances, I believe we should focus on doing a second PoC that prototypes the changes to the `ReadService` to double-check that we really understand the whole scope. What do you think?

Note that, I've already added some comments outlining required changes per kind of data to the `ReadService`. These helped me double-check that no data is missing. Feel free to add further ones.

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/reindexfromstate/StateToUpdateMapping.scala [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      ),
+      transaction = transactionFromContractState(
+        contractId,
+        contract,
+        contractInstance,
+        submissionSeed,
+      ),
+      transactionId = TransactionId.assertFromString(
+        UUID.randomUUID().toString
+      ), // TODO: Can be determined from state?
```
[the remark this answers is not in the document]
Comment (bame-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/reindexfromstate/StateToUpdateMapping.scala:152]: Yes, my mental model is that you have available:
1. Active state at pruning_end
2. Transaction log from pruning_end to ledger end

[the remark this answers is not in the document]
Comment (andreaslochbihler-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/reindexfromstate/StateToUpdateMapping.scala:152]: That's also in line with my mental model.

**Pull request closed without merging**